sound/soc/sof/topology.c

```diff
@@ -1429,7 +1429,8 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 }
 
 static int sof_widget_load_dai(struct snd_soc_component *scomp, int index,
-			       struct snd_sof_widget *swidget, int core,
```

Comment:
I have no idea what this patch does, it's completely unrelated to the commit message and I don't see why you are removing stuff here.

Reply:
The "core" is included in the "template".

Reply:
What is a 'template'?
I don't think you could have chosen a more confusing variable name.

Reply:
Thanks, just wanted to emphasize that it is only a template for the new sof_ipc_comp initialization. Let me change it to 'comp' or 'ipc_comp' if it is confusing people.

Reply:
`base` as in "base_type"
https://lwn.net/Articles/446317/ Object-oriented design patterns in the kernel, part 2

Reply:
just my 2 cents, ASoC for example is using "template" when creating new kcontrols. So might be actually familiar term for upstream...

Reply:
this change will not be applied to the new solution, mark it resolved.